mono/metadata/icall.c

```diff
@@ -2163,18 +2163,29 @@ ves_icall_MonoField_SetValueInternal (MonoReflectionFieldHandle field, MonoObjec
 		mono_gchandle_free (value_gchandle);
 }
 
+ICALL_EXPORT MonoObject*
+mono_TypedReference_ToObject (MonoTypedRef* tref);
+
 ICALL_EXPORT void
 ves_icall_System_RuntimeFieldHandle_SetValueDirect (MonoReflectionField *field, MonoReflectionType *field_type, MonoTypedRef *obj, MonoObject *value, MonoReflectionType *context_type)
 {
+	ERROR_DECL (error);
+
 	MonoClassField *f;
+	MonoObjectHandle objHandle, valueHandle;
```

Comment:
You can move these down to the init. Save lines. Lose no readability.
FooHandle foo = MONO_NEW_HANDLE (foo, …);